cime_config/SystemTests/mksurfdataesmf.py

```diff
@@ -0,0 +1,96 @@
+"""
+This test passes if mksurfdata_esmf generates an fsurdat (surface dataset)
+and the CTSM completes a simulation with this fsurdat file.
+
+We test res = '10x15' because it uses a lower-res topography file instead of
+the 1-km topography raw dataset. The 1-km file causes the test to run out of
+memory on cheyenne.
+
+Currently casper complains that `git -C` is not a valid option.
+I added -C to the `git describe` in gen_mksurfdata_namelist.py for this
+system test to work.
+"""
+
+import os
+import subprocess
+from datetime import datetime
+from CIME.SystemTests.system_tests_common import SystemTestsCommon
+from CIME.XML.standard_module_setup import *
+from CIME.SystemTests.test_utils.user_nl_utils import append_to_user_nl_files
+
+logger = logging.getLogger(__name__)
+
+class MKSURFDATAESMF(SystemTestsCommon):
+
+    def __init__(self, case):
+        """
+        initialize an object interface to the SMS system test
+        """
+        SystemTestsCommon.__init__(self, case)
+
+        # Paths and strings needed throughout
+        self._ctsm_root = self._case.get_value('COMP_ROOT_DIR_LND')
+        self._tool_path = os.path.join(self._ctsm_root, 'tools/mksurfdata_esmf')
+        _time_stamp = datetime.today().strftime("%y%m%d")
+        self._res = '10x15'  # see important comment in script's docstring
+        self._model_yr = '1850'
+        self._fsurdat_out_prefix = os.path.join(self._get_caseroot(), f'surfdata_{self._res}_hist_78pfts_CMIP6_{self._model_yr}_c{_time_stamp}.')
+
+    def build_phase(self, sharedlib_only=False, model_only=False):
+        """
+        Build executable that will generate fsurdat
+        Generate namelist for generating fsurdat
+        SKIP Generate jobscript that runs executable
+        Modify user_nl_clm to point to the generated fsurdat
+        """
+        # These steps need only happen once
+        if not os.path.exists(os.path.join(self._get_caseroot(),
+            'done_MKSURFDATAESMF_setup.txt')):
+            # Paths and strings
+            self._rm_bld_dir = f"rm -rf {self._tool_path}/bld"
```

Comment:
It's not ideal that this needs to mess with directories in the CTSM sandbox. Some issues I see with that are:
- Trying to run multiple tests of this test type could cause failures due to race conditions: the tests could stomp on each other's build directory
- You could have problems if you're running the testing from a place where you also want to actually use the mksurfdata build for some real work at the same time
- You can't run tests from a directory where you don't have write permission (may not be an issue in practice)

I think the solution to this is to run the cmake and make commands from some other location that is inside the test directory rather than being under the ctsm root. If I remember correctly, the way cmake works is that you run cmake from your desired build location, pointing to the source directory. So you could create a build directory inside the test directory then run `cmake /path/to/ctsmroot/tools/mksurfdata_esmf/src`. For this to work, the paths in gen_mksurfdata_build would need to be generalized so that it could be run from anywhere. Or maybe there's some other way to accomplish this goal.

I don't feel this absolutely needs to be done immediately, but it should be done at some point.

Reply:
As a temporary solution, the test now deletes the /tools/mksurfdata_esmf/bld directory when it's done needing it. This reduces the likelihood of the above risks without eliminating it.

To resolve this, I would like to discuss it in a meeting (me and @ekluzek ?)

Reply:
In my view, the temporary solution could make things worse in a number of situations: where some other process is expecting the build to be there, and then it gets removed out from underneath. So I'd say what you had before this was probably a bit better... but really it's hard to avoid problems in certain situations until you move to a build in a test-specific location.

Reply:
Ok, I reverted to what I had before this.